src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -137,6 +139,11 @@ private SchemaKeyspace()
               + "kind text,"
               + "position int,"
               + "type text,"
+              + "mask_keyspace text,"
+              + "mask_name text,"
+              + "mask_argument_types frozen<list<text>>,"
+              + "mask_argument_values frozen<list<blob>>,"
+              + "mask_argument_nulls frozen<list<boolean>>," // arguments that are null
```

Comment:
The schema produced by this will be rejected by Cassandra versions without this patch. So this will break downgrades. This is ok if this lands in a major (5.0), but not if it goes to a minor (4.2).

As an alternative, we can just put the masks on a separate table, this way: https://github.com/adelapena/cassandra/commit/cc9983911c6a99954a1e6ef0aeffa8f439f9b888

That separate table for masks will just be ignored by older versions during a downgrade. It replicates the primary key of `system_schema.columns`, which is not ideal unless that's done for supporting downgrades. @blerer wdyt?

Reply:
The separate table solution seems superior right? It has disadvantage and a clear advantage.

Reply:
Using two tables has disadvantages in terms of complexity, space used and performance (we duplicate reads and writes when loading and storing the schema). However, the table is small and we don't write it very often, so the benefit of not breaking backwards compatibility could very well outweigh the costs, especially since we want to put this on current trunk, which is still 4.2.

Reply:
@blerer what do you think of this? Should we go with the separate table?

Reply:
Just for the record it was commented yesterday 5.0 will be the choice granted by other changes such as removal of scripted UFFs.

Reply:
In light of the current discussions about downgradability it might be prudent to use a separate table for the masks, so the schema can be opened after an aborted upgrade. Otherwise, the new columns on the table column would fail at startup. wdyt?

Reply:
Yep I agree

Reply:
Just moved the masks to a separate table, so the new columns don't prevent downgrades.